File: Iteration2/code/CSE3063GRP12/src/features/login/LoginController.java
package features.login;

import core.repositories.UserRepository;
import features.main_menu.MenuController;
import core.exceptions.UserNotFoundException;
import core.exceptions.WrongPasswordException;
import core.general_providers.TerminalManager;

public class LoginController {
    private LoginView loginView;
    private UserRepository userRepository;
    
    // Constructor
    public LoginController() {
        this.loginView = new LoginView();
        this.userRepository = new UserRepository();
        handleLogin();
    }

    // navigate to the menu
    public void navigateToMenu() {
        new MenuController();
    }

    // get username input from the user
    public String getUserNameInput() {
        loginView.displayUsername();
        return TerminalManager.getInstance().read(); 
    }

    // get password input from the user
    public String getPasswordInput() {
        loginView.displayPassword();
        return TerminalManager.getInstance().read();
    }

    // handle the login logic
    public void handleLogin() {
        boolean loginSuccessful = false;
        while (!loginSuccessful) {
            try {
                String username = getUserNameInput();
                String password = getPasswordInput();
    
                userRepository.loginCheck(username, password);
                navigateToMenu();
                loginSuccessful = true;
            } catch (UserNotFoundException | WrongPasswordException e) {
                loginView.showError(e);
            } catch (Exception e) {
                loginView.showError(e);
                break;
            }
        }
    }
}

File: Iteration3/code/CSE3063GRP12/src/features/login/LoginController.py
from features.login.LoginView import LoginView
from core.repositories.UserRepository import UserRepository
from core.exceptions.UserNotFoundException import UserNotFoundException
from core.exceptions.WrongPasswordException import WrongPasswordException
from features.main_menu.MenuController import MenuController

class LoginController:

    def __init__(self):
        self.login_view = LoginView()
        self.user_repository = UserRepository()
        self.handle_login()

    def navigate_to_menu(self):
        MenuController()

    def get_username_input(self):
        return input("Please enter your username: ")

    def get_password_input(self):
        return input("Please enter your password: ")

    def handle_login(self):
        login_successful = False
        while not login_successful:
            try:
                username = self.get_username_input()
                password = self.get_password_input()

                self.user_repository.login_check(username, password)
                self.navigate_to_menu()
                login_successful = True
            except (UserNotFoundException, WrongPasswordException) as e:
                self.login_view.show_error(e)
            except Exception as e:
                self.login_view.show_error(e)
                break
